mesonbuild/compilers/mixins/clike.py

```diff
@@ -100,6 +100,8 @@ def to_native(self, copy: bool = False) -> T.List[str]:
         # Remove system/default include paths added with -isystem
         default_dirs = self.compiler.get_default_include_dirs()
         if default_dirs:
+            is_default = lambda opt_dir: any(map(lambda default_dir: opt_dir == default_dir or os.path.samefile(opt_dir, default_dir),
+                                                 default_dirs))
```

Comment:
Don’t assign lambdas like this, it’s not idiomatic in python, and there are performance considerations (and does other corner cases), use a def

also avoid filter and map in general, and prefer a comprehension